### ISSUE TITLE
Way to Build Flatpaks Behind SSL Proxy?

### ISSUE THREAD

**Issue by zicklag** [flatpak/flatpak#3746]
## Linux distribution and version

Pop!_OS 20.04 LTS

## Flatpak version

flatpak 1.6.3

## Description of the problem

I am unable to build flatpaks behind an HTTP proxy with SSL Bump ( i.e. you have to trust a custom certificate ). I have installed the certificate on the system by putting the cert in `/usr/local/share/ca-certificates/proxyca.crt` and running `update-ca-certificates`, which works for most programs on the system. The issue is that when the flatpak tries to download the archives that are defined in the manifest, it fails because of the untrusted certificate:

## Steps to Reproduce

1. Set the `http_proxy` and `https_proxy` environment variables properly to connect to an HTTP proxy that is replacing the HTTPS certificates with it's own custom certificate.
2. Add the proxy certificate to `/usr/local/share/ca-certificates/proxyca.crt`
3. Run `update-ca-certificates`

Run these commands:

```
$ git clone https://github.com/flathub/org.gnome.Dictionary.git
$ cd org.gnome.Dictonary
$ flatpak-builder build-dir org.gnome.Dictionary.yml
Downloading sources
Downloading https://download.gnome.org/sources/gnome-dictionary/3.26/gnome-dictionary-3.26.1.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

Failed to download sources: module gnome-dictionary: server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
```

---

Is there anyway to add certificates to flatpak-builder's trust so that I can build the flatpaks behind the proxy?

**Comment by zicklag** [comment]
I just realized the issue is probably that I installed `flatpak-builder` with Flatpak, but it isn't respecting my extra CA cert. How do you setup CA certs that Flatpaks can find?

**Comment by zicklag** [comment]
If it makes a difference I can see my custom certificate when I run `trust list`.

**Comment by zicklag** [comment]
By running the following command I can confirm that `curl` inside the `flatpak-builder` flatpak is not sharing my trusted certificates, which causes the failure to download the assets. I suppose one way around this might be to install flatpak builder without flatpak. I'll try that first.

    flatpak run --share=network --command=bash org.flatpak.Builder -l

**Comment by zicklag** [comment]
Looks like the issue is the `libcurl` config in the Freedesktop SDK: https://github.com/flatpak/flatpak/issues/2721#issuecomment-660148315.

**Comment by zicklag** [comment]
I opened an [issue](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1087) with the Freedesktop SDK.

Also installing `flatpak-builder` outside of a flatpak using [`nix`](https://nixos.org) has been a decent workaround for me for now.

**Comment by nanonyme** [comment]
Wait, so is this org.flatpak.Builder issue? It can probably be workarounded within that app.

**Comment by zicklag** [comment]
Yeah, it was a problem with the builder. The org.flatpak.Builder flatpak needed to be configured so that it would respect the trusted certs on the host.

**Comment by nanonyme** [comment]
How was this resolved in the end? I see this issue is still open and I don't see anything suspect in https://github.com/flathub/org.flatpak.Builder/commits/master

**Comment by zicklag** [comment]
I don't know that it is fixed, but I can try it again. My solution was just to install the `flatpak-builder` command with the Nix package manager which didn't have the same problem with libcurl and trusting the host certs.

---

Yep, I tried it and I'm still getting the error:

```
Emptying app dir 'build'
Downloading sources
fatal: unable to access 'https://github.com/mirror/x264.git/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
Failed to download sources: module x264: Child process exited with code 128
```

**Comment by nanonyme** [comment]
@alexlarsson any idea if we could make flatpak/flatpak-builder tell libcurl to pick GnuTLS?
